README.md

```diff
@@ -2217,4 +2217,88 @@ A Japanese word meaning "form", relating to karate. Used commonly in programming
 
 An excellent example of a Test Driven Development Kata:
 
-[The Bowling Game](http://butunclebob.com/ArticleS.UncleBob.TheBowlingGameKata)
\ No newline at end of file
+[The Bowling Game](http://butunclebob.com/ArticleS.UncleBob.TheBowlingGameKata)
+
+## Coding in the Command Line
+
+The Command Line is an example of a Read-Evaluate-Print-Loop (REPL). Users can enter expressions which will be read, evaluated and the answer printed to the screen. The environment will then return to the read state. This provides the ability to enter single expressions in different programming languages (once you have installed the program and called it from the command line). This ability can be useful for testing simple concepts or debugging.
+
+For example, if you write `python` into the command line and press enter you can then write the following:
+
+``` python
+def fib(n):
+    if n<3:
+        return n
+    return fib(n-1) + fib(n-2)
+```
+
+Getting the indentations right is very important as Python does not have brackets for if statements.
+
+You can then enter `fib(4)` and the command line will return `5`, or `fib(8)` and `34` will be returned.
+
+## Some different coding languages and their usages
+
+### MatLab (Matrix Laboratory)
+Good for matrices and vectors, which are the key components of linear algebra.
+
+### R
+Good for Statistics.
+
+### Python
+Good for anything, but especially useful for Machine Learning. Python had a major release to 3.0 at the end of 2008, but some software still uses 2.6 or 2.7 as it cannot handle the breaking changes.
+
+### Java
+Good for anything. Originally comes from Oracle Java, but was taken by developers and made into the open-source Java Development Kit (JDK). Oracle Java is almost the same as JDK.
+
+### Access to Software
+
+Some software is proprietary, others are open source and some are even free.
+
+#### Proprietary
+
+Owned by a company. Users are required to purchase a license. Examples include MatLab.
+
+#### Open Source
+
+The source code is made available by the copyright holder under a license that allows users to change and distribute the code. Exampes include Python and OpenJDK.
+
+#### Free
+
+Free as in freedom, not free as in beer. The code is made available for users to change and distribute. Examples include the Linux kernal.
+
+## NoSQL Databases
+
+NoSQL generally means anything but SQL, although some NoSQL databases can support SQL (e.g. postgreSQL).
+
+- Mongo. Documents are written in JSON. Can have as many fields/columns as you want. Allows as many connections as needed.
+
+- Cassandra. Similar to Mongo.
+
+- Redis. Similar to Maps in Java in that it is made up of keys and values. Processes are performed sequentially rather than all at once, but as it is done in RAM it is very performant. Good at keeping data in RAM and retrieving previously calculated values that won't change. High risk of errors if values can change and need to be recalculated.
```

Comment:
Not sure what that errors reference was to. I think that might have been part of a slightly different topic (in-app local caches stored in e.g. a HashMap).

```suggestion
- Redis. Similar to Maps in Java in that it is made up of keys and values. Processes are performed sequentially rather than concurrently, but as it is done in RAM it is still very performant. Good at keeping data in RAM and retrieving previously calculated values that won't change.
```